command/server.go

```diff
@@ -1259,47 +1291,59 @@ func (c *ServerCommand) Run(args []string) int {
 	var ok bool
 	if config.HAStorage != nil {
 		if config.Storage.Type == storageTypeRaft && config.HAStorage.Type == storageTypeRaft {
-			c.UI.Error("Raft cannot be set both as 'storage' and 'ha_storage'. Setting 'storage' to 'raft' will automatically set it up for HA operations as well")
+			err = fmt.Errorf("Raft cannot be set both as 'storage' and 'ha_storage'. Setting 'storage' to 'raft' will automatically set it up for HA operations as well")
+			observer.Error("config-ha-storage", err)
+			c.UI.Error(err.Error())
 			return 1
 		}
 
 		if config.Storage.Type == storageTypeRaft {
-			c.UI.Error("HA storage cannot be declared when Raft is the storage type")
+			err = fmt.Errorf("HA storage cannot be declared when Raft is the storage type")
+			observer.Error("config-ha-storage", err)
+			c.UI.Error(err.Error())
 			return 1
 		}
 
 		factory, exists := c.PhysicalBackends[config.HAStorage.Type]
 		if !exists {
-			c.UI.Error(fmt.Sprintf("Unknown HA storage type %s", config.HAStorage.Type))
+			err = fmt.Errorf(fmt.Sprintf("Unknown HA storage type %s", config.HAStorage.Type))
+			observer.Error("config-ha-storage", err)
+			c.UI.Error(err.Error())
 			return 1
-
 		}
 
 		namedHALogger := c.logger.Named("ha." + config.HAStorage.Type)
 		c.allLoggers = append(c.allLoggers, namedHALogger)
 		habackend, err := factory(config.HAStorage.Config, namedHALogger)
 		if err != nil {
-			c.UI.Error(fmt.Sprintf(
-				"Error initializing HA storage of type %s: %s", config.HAStorage.Type, err))
+			err = fmt.Errorf(
```

Comment:
if you use the `%w` verb for the error in an `fmt.Errorf()` then it can be unwrapped